sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -563,3 +590,140 @@ void AttestationAdministrationClient::RetrieveResponseValidationCollateral(
+
+  AttestationTokenValidationOptions tokenOptions = options.TokenValidationOptionsOverride
+      ? *options.TokenValidationOptionsOverride
+      : m_parentClient->m_tokenValidationOptions;
+
+  auto deferredShared(
+      std::make_shared<AttestationBatchShared<Models::AttestationToken<Models::PolicyResult>>>(
+          request,
+          [this, tokenOptions](std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse) {
+            switch (rawResponse->GetStatusCode())
+            {
+
+              // 200, 201, 202, 204 are accepted responses
+              case Azure::Core::Http::HttpStatusCode::Ok:
+              case Azure::Core::Http::HttpStatusCode::Created:
+              case Azure::Core::Http::HttpStatusCode::Accepted:
+              case Azure::Core::Http::HttpStatusCode::NoContent:
+                break;
+              default:
+                throw Azure::Core::RequestFailedException(rawResponse);
+            }
+            return m_parentClient->ProcessSetPolicyResponse(tokenOptions, rawResponse);
+          },
+          sharedContext));
+
+  std::shared_ptr<DeferredResponseSharedBase> sharedBase(deferredShared);
+  return CreateDeferredResponse<Models::AttestationToken<Models::PolicyResult>>(sharedBase);
+}
+
+/**
+ * @brief Resets the attestation policy for the specified AttestationType to its default.
+ *
+ * @param attestationType Sets the policy on the specified AttestationType.
+ * @param options Options used when setting the policy, including signer.
+ * @return Response<Models::AttestationToken<Models::PolicyResult>> The result of the reset
+ * policy operation.
+ *
+ * @note \b Note: The RetrieveResponseValidationCollateral API \b MUST be called before the
+ * ResetAttestationPolicy API is called to retrieve the information needed to validate the
+ * result returned by the service.
+ */
+Azure::Core::DeferredResponse<Models::AttestationToken<Models::PolicyResult>>
+AttestationBatchFactory::ResetAttestationPolicy(
+    Models::AttestationType const& attestationType,
+    SetPolicyOptions const& options)
+{
+  // Calculate a signed (or unsigned) attestation policy token to send to the service.
+  Models::AttestationToken<void> const tokenToSend(m_parentClient->CreateAttestationPolicyToken(
+      Azure::Nullable<std::string>(), options.SigningKey));
+
+  auto sharedContext = std::make_shared<
+      AttestationBatchShared<Models::AttestationToken<Models::PolicyResult>>::SharedContext>(
+      tokenToSend.RawToken);
+
+  Azure::Core::Http::Request request
+      = m_parentClient->CreateResetPolicyRequest(attestationType, sharedContext->BodyStream);
+
+  AttestationTokenValidationOptions tokenOptions = options.TokenValidationOptionsOverride
+      ? *options.TokenValidationOptionsOverride
+      : m_parentClient->m_tokenValidationOptions;
+
+  auto deferredShared(
+      std::make_shared<AttestationBatchShared<Models::AttestationToken<Models::PolicyResult>>>(
+          request,
```

Comment:
In storage SDK, we have protocol layer functions and convenience layer functions.

Protocol layer function is generated by code-gen. It constructs `Azure::Core::Http::Request` as local variable, and send the request over the wire. [Here](https://github.com/Azure/azure-sdk-for-cpp/blob/6d102d09ae416cb137c99fe11e1a21fced793eb3/sdk/storage/azure-storage-blobs/src/rest_client.cpp#L3956) is an example of DeleteBlob protocol layer function.

Convenience layer function is a wrapper on top of protocol layer function, and provide more user-friendly API interface.

Convenience layer function doesn't have access to `Core::Http::Request` the entire time. So there's not way we can call a constructor like this.

Reply:
This is a proof of concept of an implementation of the preferred batch pattern. I tried to make it clear in the DeferredOperation.md that this was a possible implementation of the pattern, but not a required implementation.

The only *required* part of this pattern is the public API surface, specifically:
1. The `DeferredResponse<T>` type with a public `GetResponse()` method; 
1. A `CreateBatchFactory()` method on the client class and 
1. A `SubmitBatch()` method on the client class which takes a `BatchFactory` object as a parameter.
   1. The `SubmitBatch()` throws all network layer exceptions, the `GetResponse()` method throws any exceptions returned by the service.

Everything else you see here is a part of a proof-of-concept prototype to ensure that the pattern could be reasonably implemented.